### PR TITLE
Notes accessibility

### PR DESCRIPTION
**Pull request by rowanhemsi**

### CODE REVIEW

**Review by rowanhemsi — COMMENTED** [review]
Review with @alexander-newton and @jaetowers

File: gptables/examples/iris.py [hunk 1]
```diff
@@ -40,17 +40,12 @@
 title = "Mean Iris $$note2$$ sepal dimensions"
```
[the remark this answers is not in the document]
Comment (rowanhemsi) [reply gptables/examples/iris.py:40]: Separate user story for this

File: gptables/core/contentsheet.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
+from dataclasses import dataclass
```
[the remark this answers is not in the document]
Comment (rowanhemsi) [reply gptables/core/contentsheet.py]: make_table_of_contents method added

File: gptables/core/api.py [hunk 3]
```diff
@@ -48,11 +50,24 @@ def produce_workbook(
...
+
+    if contentsheet is not None:
+        ws = wb.add_worksheet(contentsheet.label)
+        ws.write_contentsheet(contentsheet, auto_width)
     
+    # Add notesheet in correct position
+    if notesheet is not None:
+        ws = wb.add_worksheet(notesheet.label)
+
     for sheet, gptable in sheets.items():
```
[the remark this answers is not in the document]
Comment (rowanhemsi) [reply gptables/core/api.py]: Contents and notes sheets added in loop. Cover added separately as it does not contain a table

File: gptables/examples/iris.py [hunk 4]
```diff
@@ -61,17 +56,29 @@
...
 ## Define our GPTable
 iris_table = gpt.GPTable(table=iris_summary, **kwargs)
 
+sheets = {"Iris Flower Dimensions": iris_table}
+
+## Notesheet
+notes = {
+    "Note reference": ["note1", "note2"],
+    "Note text": ["I've got 99 problems and taxonomy is one.", "Goo Goo Dolls, 1998."],
+    "Useful link": ["google", "duckduckgo"],
```
[the remark this answers is not in the document]
Comment (rowanhemsi) [reply gptables/examples/iris.py]: Markdown link formatting added for tables and text elements

File: gptables/__init__.py [hunk 5]
```diff
@@ -1,6 +1,6 @@
 from gptables.core.theme import Theme
 from gptables.core.cover import Cover
-from gptables.core.gptable import GPTable
+from gptables.core.gptable import GPTable, FormatList # FormatList for testing
```
Comment (Jacobb164) [inline gptables/__init__.py:3]: Does FormatList need to still be in here if it was for testing?

File: gptables/core/gptable.py [hunk 6]
```diff
@@ -1,4 +1,5 @@
 import pandas as pd
+import re
```
Comment (Jacobb164) [inline gptables/core/gptable.py:2]: Does "re" this need to be added to requirements / setup?

File: gptables/core/gptable.py [hunk 7]
```diff
@@ -172,7 +167,7 @@ def _valid_column_index(self, column_index):
         return column_index in range(self.table.shape[1])
         
 
-    def _set_column_headings(self):
+    def _set_column_headings(self): # TODO: check custom formatting in headers
```
Comment (Jacobb164) [inline gptables/core/gptable.py:170]: if this means using your own theme to change table headers format then this works fine.

**Review by Jacobb164 — COMMENTED** [review]
I am happy the code is working as intended. 

The iris_personal_theme examples is no longer working
the R examples need updating for new features
I looked at the TODOs you had in the code and left comments
I left a few other comments nothing major 

Happy this is working and for it to be merged

File: gptables/__init__.py (hunk 5 above)
Comment (rowanhemsi) [reply gptables/__init__.py:3]: Imported directly from module in test_wrappers, can be removed from __init__.py

File: gptables/core/gptable.py (hunk 6 above)
Comment (rowanhemsi) [reply gptables/core/gptable.py:2]: `re` is an inbuilt module, so does not need to be added as a requirement

File: gptables/core/gptable.py (hunk 7 above)
Comment (rowanhemsi) [reply gptables/core/gptable.py:170]: That's great that the theme is still working correctly. The issue is if people want to apply different formatting to specific words in the column headings. There is a WIP branch looking at this issue: [units-custom-formatting](https://github.com/best-practice-and-impact/gptables/tree/units-custom-formatting)

File: gptables/core/wrappers.py [hunk 8]
```diff
@@ -91,22 +56,29 @@ def write_gptable(self, gptable, auto_width, disable_footer_parentheses):
...
+            raise ValueError(msg)
+        
         theme = self.theme
 
         # Write each GPTable element using appropriate Theme attr
         pos = [0, 0]
 
-        self._reference_annotations(gptable)
+        self._reference_annotations(gptable, reference_order)
+        self._parse_urls(gptable) # TODO: document and raise error - attribute can't have custom formatting and url in same element
```
[the remark this answers is not in the document]
Comment (rowanhemsi) [reply gptables/core/wrappers.py:79]: Would be good to talk in person about this because I didn't think this would be possible given the package design

**Comment by rowanhemsi** [comment]
> nI am happy the code is working as intended.
> 
> The iris_personal_theme examples is no longer working the R examples need updating for new features I looked at the TODOs you had in the code and left comments I left a few other comments nothing major
> 
> Not sure if the "quick and dirty" example should be removed all together?
> 
> Happy this is working and for it to be merged

Thanks for your review @Jacobb164
Comments acted on or replied to. Stories in the backlog for broken examples and "quick and dirty" method

**Pull request merged**